react-hot-button-component/src/index.jsx

```diff
@@ -1,34 +1,36 @@
 import React, { useState } from 'react';
 import ReactDOM from 'react-dom/client';
 
-const buttonStyle = {
-  color: 'blue'
-};
-
 function HotButton() {
   const [index, setIndex] = useState(0);
   function handleClick() {
     const newIndex = index + 1;
     setIndex(newIndex);
-    if (newIndex < 3) {
-      buttonStyle.color = 'blue';
-    } else if (newIndex < 6) {
-      buttonStyle.color = 'purple';
-    } else if (newIndex < 9) {
-      buttonStyle.color = 'light-purple';
-    } else if (newIndex < 12) {
-      buttonStyle.color = 'peach';
-    } else if (newIndex < 15) {
-      buttonStyle.color = 'orange';
-    } else if (newIndex < 18) {
-      buttonStyle.color = 'yellow';
-    } else if (newIndex >= 18) {
-      buttonStyle.color = 'white';
+  }
+
+  let color = 'blue';
+  function checkIndex() {
```

Comment:
I'd suggest just putting this logic inline instead of putting it in a separate function. It's more succinct and works great. As follows:
```
let color = 'blue';
if (index < 3) {
  color = 'blue';
} else if (index < 6) {
  color = 'purple';
} else if (index < 9) {
  color = 'light-purple';
} else if (index < 12) {
  color = 'peach';
} else if (index < 15) {
  color = 'orange';
} else if (index < 18) {
      color = 'yellow';
} else { // Note you can just use `else` here instead of `else if`
  color = 'white';
}
```
So there's no need to define or call a function.